cloudant-http/src/main/java/com/cloudant/http/interceptors/CookieInterceptor.java

```diff
@@ -79,86 +84,109 @@ public HttpConnectionInterceptorContext interceptRequest(HttpConnectionIntercept
+                        HttpURLConnection.HTTP_UNAUTHORIZED) {
+                    InputStream errorStream = connection.getErrorStream();
+                    try {
+                        switch (statusCode) {
+                            case HttpURLConnection.HTTP_FORBIDDEN: //403
+                                //check if it was an expiry case
+                                String errorString = IOUtils.toString(errorStream, "UTF-8");
+                                // Check using a regex to avoid dependency on a JSON library.
+                                // Note (?siu) flags used for . to also match line breaks and for
+                                // unicode
+                                // case insensitivity.
+                                if (!errorString.matches("(?siu)" +
+                                        ".*\\\"error\\\"\\s*:\\s*\\\"credentials_expired\\\".*")) {
+                                    // Wasn't a credentials expired, throw exception
+                                    HttpConnectionInterceptorException toThrow = new
+                                            HttpConnectionInterceptorException(errorString);
+                                    // Set the flag for deserialization
+                                    toThrow.deserialize = true;
+                                    throw toThrow;
+                                } else {
+                                    // Was expired - set boolean to renew cookie
+                                    renewCookie = true;
+                                }
+                                break;
+                            case HttpURLConnection.HTTP_UNAUTHORIZED: //401
+                                IOUtils.toString(errorStream, "UTF-8");
+                                // We need to get a new cookie
                                 renewCookie = true;
-                            }
-                            break;
-                        case HttpURLConnection.HTTP_UNAUTHORIZED: //401
-                            IOUtils.toString(errorStream, "UTF-8");
-                            // We need to get a new cookie
-                            renewCookie = true;
-                            break;
-                        default:
-                            break;
+                                break;
+                            default:
+                                break;
+                        }
+                    } finally {
+                        IOUtils.closeQuietly(errorStream);
+                    }
+                    if (renewCookie) {
+                        boolean success = requestCookie(connection.getURL(), context);
+                        // Don't resend request, failed to get cookie
+                        if (success) {
+                            context.replayRequest = true;
+                        } else {
+                            // Didn't successfully renew, maybe creds are expired
```

Comment:
The thought I had in my mind was expired in the sense of having been changed on the server since this is a renewal and they must have been valid previously if we are trying to renew, but I see your point! Fixed in b1ef8a8.